packages/wdio-cucumber-framework/src/reporter.js

```diff
@@ -0,0 +1,272 @@
+import { Status } from 'cucumber'
+
+import CucumberEventListener from './cucumberEventListener'
+import { createStepArgument } from './utils'
+import * as path from 'path'
+
+class CucumberReporter {
+    gherkinDocEvents = []
+
+    constructor (eventBroadcaster, options, cid, specs, reporter) {
+        this.capabilities = options.capabilities
+        this.tagsInTitle = options.tagsInTitle || false
+        this.options = options
+        this.cid = cid
+        this.specs = specs
+        this.reporter = reporter
+        this.failedCount = 0
+
+        this.eventListener = new CucumberEventListener(eventBroadcaster)
+            .on('before-feature', this.handleBeforeFeature.bind(this))
+            .on('before-scenario', this.handleBeforeScenario.bind(this))
+            .on('before-step', this.handleBeforeStep.bind(this))
+            .on('after-step', this.handleAfterStep.bind(this))
+            .on('after-scenario', this.handleAfterScenario.bind(this))
+            .on('after-feature', this.handleAfterFeature.bind(this))
+    }
+
+    handleBeforeFeature (uri, feature) {
+        this.featureStart = new Date()
+
+        this.emit('suite:start', {
+            uid: this.getUniqueIdentifier(feature),
+            title: this.getTitle(feature),
+            type: 'suite',
+            file: uri,
+            tags: feature.tags,
+            description: feature.description,
+            keyword: feature.keyword
+        })
+    }
+
+    handleBeforeScenario (uri, feature, scenario) {
+        this.scenarioStart = new Date()
+        this.testStart = new Date()
+
+        this.emit('suite:start', {
+            uid: this.getUniqueIdentifier(scenario),
+            title: this.getTitle(scenario),
+            parent: this.getUniqueIdentifier(feature),
+            type: 'suite',
+            file: uri,
+            tags: scenario.tags
+        })
+    }
+
+    handleBeforeStep (uri, feature, scenario, step, sourceLocation) {
+        this.testStart = new Date()
+
+        this.emit('test:start', {
+            uid: this.getUniqueIdentifier(step),
+            title: step.text,
+            type: 'test',
+            file: uri,
+            parent: this.getUniqueIdentifier(scenario, sourceLocation),
+            duration: new Date() - this.testStart,
+            tags: scenario.tags,
+            featureName: feature.name,
+            scenarioName: scenario.name,
+            argument: createStepArgument(step)
```

Comment:
are we adding argument for reporters?

If yes, then which reporters? I'm asking because this value is currently ignored by allure reporter. 
Also, it expects argument name and value, but we can set it arg1 or whatever

Reply:
DataTable is huge, it can't be added as Allure argument, only as an attachment.

![image](https://user-images.githubusercontent.com/25589559/60819087-399d6880-a19f-11e9-83b5-1f194de06499.png)

Reply:
anyway it is not blocking this PR

Reply:
Yes we are adding it so that all reporters can use the data. Since we just migrated the framework, no reporter is using it but we should add support for it in the next PRs